plugins/jvm-example/src/main/java/org/elasticsearch/plugin/example/JvmExamplePlugin.java

```diff
@@ -111,5 +117,16 @@ public Settings additionalSettings() {
     public void onModule(RepositoriesModule repositoriesModule) {
     }
 
-
+    /**
+     * Module decalaring some example configuration and a _cat action that uses
+     * it.
+     */
+    public static class ConfiguredExampleModule extends AbstractModule {
+        @Override
+        protected void configure() {
+          bind(ExamplePluginConfiguration.class).asEagerSingleton();
+          Multibinder<AbstractCatAction> catActionMultibinder = Multibinder.newSetBinder(binder(), AbstractCatAction.class);
```

Comment:
Having limitless extensibility is not a good thing... as a plugin developer, I want to know what I can and cannot do... what I can extend. Otherwise I can easily do something I really shouldn't, break something along the way, without even knowing I broke it. Having well defined extension points and effectively limiting the extensibility of es in general:

* helps us make sure plugins cannot break things (as they'll be restricted to what we allow them to do)
* helps the users know what they can do and rest assured that they're not doing something they're not supposed to

So overall, personally I'm less concerned about not capturing all the extension points at first run.. I'm more concerned about first capturing control over it. Then we can start opening up extension points as we see fit in a controlled manner.